tools/dissector/sample_dissector.cpp

```diff
@@ -196,37 +193,94 @@ namespace OpenDDS
       return ns_stack_.back();
     }
 
-    void Sample_Base::add_protocol_field(enum ftenum ft, field_display_e fd) {
-      Field_Context * fc = get_context();
-      Sample_Manager::instance().add_protocol_field(
-        &fc->hf_, get_ns(), fc->label_, ft, fd
-      );
+    void Sample_Base::push_idl_name(std::string name) {
+      Sample_Base::clear_ns();
+
+      // If name begins with "IDL:", remove it
+      const std::string idl_prefix("IDL:");
+      if (name.substr(0, idl_prefix.size()) == idl_prefix) {
+          name.erase(0, idl_prefix.size());
+      }
+
+      // If name contains ':', remove everything after the last ':'
+      size_t l = name.rfind(":");
+      if (l != std::string::npos) {
+          name.erase(l, name.size() - l);
+      }
+
+      // Push namespace when we find a '/'
+      name.push_back('/'); // For last namespace
+      l = 0;
+      for (size_t index = 0; index != name.size(); index++) {
+          if (name[index] == '/') {
+              Sample_Base::push_ns(name.substr(index - l, l));
+              l = 0;
+          } else {
+              l++;
+          }
+      }
+    }
+
+    void Sample_Base::add_protocol_field() {
+      Field_Context* fc = get_context();
+      if (fc) {
+        Sample_Manager::instance().add_protocol_field(fc->hf_info_);
+      }
     }
 
-    Sample_Field::Sample_Field (IDLTypeID id, const std::string &label)
-      : label_ (label),
-        type_id_ (id),
-        nested_ (0),
+    Field_Context* Sample_Base::create_context(ftenum ft, field_display_e fd) {
+      if (field_contexts_.count(get_ns())) {
+        throw Sample_Dissector_Error(
+          std::string("Could not create context becuase it already exists for: ") +
+          get_ns()
+        );
+      }
+
+      Field_Context* fc = new Field_Context;
+      fc->long_name_ = get_ns();
+      fc->short_name_ = get_label();
+      fc->hf_ = -1;
+      fc->hf_info_ = { &fc->hf_, {
+      // Note: Wireshark calls this second struct member "abbrev" (What we
```

Comment:
See if this comment can be reworded to make it more clear ("Wireshark calls" and "is the string" is a bit clunky).
You can remove the line "This was confusing to me...".